src/visitors/cvode_visitor.cpp

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright 2023 Blue Brain Project, EPFL.
+ * See the top-level LICENSE file for details.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "visitors/cvode_visitor.hpp"
+
+#include "ast/all.hpp"
+#include "lexer/token_mapping.hpp"
+#include "pybind/pyembed.hpp"
+#include "utils/logger.hpp"
+#include "visitors/visitor_utils.hpp"
+#include <optional>
+#include <utility>
+
+namespace pywrap = nmodl::pybind_wrappers;
+
+namespace nmodl {
+namespace visitor {
+
+static int get_index(const ast::IndexedName& node) {
+    return std::stoi(to_nmodl(node.get_length()));
+}
+
+static void remove_conserve_statements(ast::StatementBlock& node) {
+    auto conserve_equations = collect_nodes(node, {ast::AstNodeType::CONSERVE});
+    if (!conserve_equations.empty()) {
+        std::unordered_set<ast::Statement*> eqs;
+        for (const auto& item: conserve_equations) {
+            eqs.insert(std::dynamic_pointer_cast<ast::Statement>(item).get());
+        }
+        node.erase_statement(eqs);
+    }
+}
+
+static std::pair<std::string, std::optional<int>> parse_independent_var(
+    std::shared_ptr<ast::Identifier> node) {
+    auto variable = std::make_pair(node->get_node_name(), std::optional<int>());
+    if (node->is_indexed_name()) {
+        variable.second = std::optional<int>(
+            get_index(*std::dynamic_pointer_cast<const ast::IndexedName>(node)));
+    }
+    return variable;
+}
+
+/// set of all indexed variables not equal to ``name``
+static std::unordered_set<std::string> get_indexed_variables(const ast::Expression& node,
+                                                             const std::string& name) {
```

Comment:
the name still hasn't improved. this can easily be made trivial to understand by using a better name.

Reply:
Not really sure what's a better name for getting indexed variables than `get_indexed_variables`, but I am open to suggestions.

Reply:
Then I think I misunderstood the original comment 😅 Just to clarify: the variable `name` should be `ignored_name`, correct?